services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -147,9 +151,36 @@ public UserRepresentation account() {
 
         rep.setAttributes(profile.getAttributes().getReadable(false));
 
+        if(userProfileMetadata == null || userProfileMetadata.booleanValue())
+            rep.setUserProfileMetadata(createUserProfileMetadata(profile));
+        
         return rep;
     }
+    
+    private UserProfileMetadata createUserProfileMetadata(final UserProfile profile) {
+        Map<String, List<String>> am = profile.getAttributes().getReadable();
+        
+        if(am == null)
+            return null;
+        
+        List<UserProfileAttributeMetadata> attributes = am.keySet().stream()
+                                                          .map(name -> profile.getAttributes().getMetadata(name))
+                                                          .filter(Objects::nonNull)
+                                                          .sorted((a,b) -> Integer.compare(a.getGuiOrder(), b.getGuiOrder()))
+                                                          .map(sam -> toRestMetadata(sam, profile))
+                                                          .collect(Collectors.toList());  
+        return new UserProfileMetadata(attributes);
+    }
 
+    private UserProfileAttributeMetadata toRestMetadata(AttributeMetadata am, UserProfile profile) {
+        return new UserProfileAttributeMetadata(am.getName(), 
+                                                am.getAttributeDisplayName(), 
+                                                profile.getAttributes().isRequired(am.getName()), 
+                                                profile.getAttributes().isReadOnly(am.getName()), 
+                                                am.getAnnotations(), 
+                                                am.getValidators() == null ? null : am.getValidators().stream().filter(avm -> !avm.getValidatorId().startsWith("up-")).collect(Collectors.toMap(AttributeValidatorMetadata::getValidatorId, AttributeValidatorMetadata::getValidatorConfig)));
```

Comment:
I though about it before, but I'm not sure if this should be in Validator SPI itself. This mechanism for hiding is really internal, specific for user profile only, not sure if it should be in generic SPI which can provide validators for other use cases. If yes then it should be some form of more generic flagging probably rather than one true/false method.

Another approach might be use of marking interface for these validators. But drawback of this solution (same as for the previous solution) is that it will be necessary to really get validator from SPI to check if it should be put into metadata. This adds extra processing and will slow down REST responses each time metadata are collected (probably not too much, but still).

This is why I implemented mechanism based on the naming convention of the `validatorId` which is there already, without need to query Validator SPI at all.
If you don't like this approach, we should maybe use validator configuration, as it is also available for the processing without need to query SPI. For user profile internal validators we can put some key into the cinfig and filter metadata for the REST API based on it's presence. We can maybe even document this as a feature how to prevent some validator configured in UserProfile JSON to be announced in REST API.

Reply:
Implemented configuration based approach to hide validators from REST Account API metadata.
Failing test doesn't look to be related, but I'm not fully sure what it tests. Could you check please?

Reply:
I rerun tests and they are all OK now

Reply:
I do not think it is specific for user profile but something that might be useful once we start using more validators where some of them are supposed to be used only internally.

One example is a validator for running client validation. Which we should be working very soon to replace with the validation spi.

Reply:
And what is problem with that validator? Not to be shown in the selection of validators available for attributed in User Profile Admin GUI? This is currently done by `ConfiguredProvider` - only validators which have this interface implemented are shown in the Admin GUI.

Reply:
That makes more sense, let us use then that interface in the check.

Reply:
Makes sense, PR updated